database/factories/ExamFactory.php

```diff
@@ -17,7 +17,13 @@ class ExamFactory extends Factory
     public function definition(): array
     {
         return [
-            //
+            'code' => $this->faker->word,
+            'grade' => $this->faker->numberBetween(10,50),
+            'credit_points' => $this->faker->numberBetween(2,8),
+            'semester' => $this->faker->numberBetween(1,6),
+            'module_id' => $this->faker->numberBetween(1,15),
```

Comment:
Since this is a foreign key, you should retrieve the id value of any module here. Otherwise it will fail in some cases.

For example this:

`Module::query()->first() ?? Module::factory()->createOne()->id`